### ISSUE TITLE
gnupg21, gpg-agent crashing on 'assertion failed'

### ISSUE THREAD

**Issue by lunaryorn** [Homebrew/homebrew-versions#1255]
gnupg21 fails to get secret keys from gpg-agent, because gpg-agent crashes after pinentry interaction.  

I can consistently reproduce this issue with:

```
$ gpg-agent --no-detach --daemon --debug-all --debug-level expert --debug-pinentry --log-file gpg2-debug.log
```

And then on another terminal:

```
hello world
gpg: signing failed: End of file
�������x����3O
                          %1����H���W(�/�I�gpg: signing failed: End of file
```

As you can see signing fails.  This happens *after* I'm asked for my passphrase.

The cause is `gpg-agent` crashing with the following message:

```
Assertion failed: (stream->flags.writing), function es_flush, file estream.c, line 1750.
```

The debug log is lengthy, but doesn't seem to have any notable messages:

```
2016-05-07 13:02:02 gpg-agent[1977] listening on socket '/Users/swiesner/.gnupg/S.gpg-agent'
2016-05-07 13:02:02 gpg-agent[1978] gpg-agent (GnuPG) 2.1.12 started
2016-05-07 13:02:05 gpg-agent[1978] handler 0x700000081000 for fd 6 started
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK Pleased to meet you
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- RESET
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION ttyname=/dev/ttys004
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION ttytype=xterm-256color
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION display=/private/tmp/com.apple.launchd.ou3UgURSwF/org.macosforge.xquartz:0
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION lc-ctype=de_DE.UTF-8
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION lc-messages=de_DE.UTF-8
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- GETINFO version
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> D 2.1.12
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION allow-pinentry-notify
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- OPTION agent-awareness=2.1.0
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- AGENT_ID
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> ERR 67109139 Unknown IPC command <GPG Agent>
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- HAVEKEY 9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0 F7A344242817D5BA4A3B6BFE0934238B87ABAC2E
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 <- KEYINFO 9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0
2016-05-07 13:02:05 gpg-agent[1978] DBG: agent_get_cache '9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0' (mode 2) ...
2016-05-07 13:02:05 gpg-agent[1978] DBG: ... miss
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> S KEYINFO 9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0 D - - - P - - -
2016-05-07 13:02:05 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- RESET
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- SIGKEY 9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- SETKEYDESC Please+enter+the+passphrase+to+unlock+the+OpenPGP+secret+key:%0A%22Sebastian+Wiesner+(Social+address)+<me@lunaryorn.com>%22%0A4096-bit+RSA+key,+ID+5C42FE98,%0Acreated+2016-04-11.%0A
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- SETHASH 8 3E01179ED8623B5A4AAFEDBDFB840B5039FDB227A252CBFB8BD37E4F9F4C9CED
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 -> OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- PKSIGN
2016-05-07 13:02:10 gpg-agent[1978] DBG: agent_get_cache '9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0' (mode 2) ...
2016-05-07 13:02:10 gpg-agent[1978] DBG: ... miss
2016-05-07 13:02:10 gpg-agent[1978] starting a new PIN Entry
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK Pleased to meet you, process 1978
2016-05-07 13:02:10 gpg-agent[1978] DBG: connection to PIN entry established
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION grab
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION ttyname=/dev/ttys004
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION ttytype=xterm-256color
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION lc-ctype=de_DE.UTF-8
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION lc-messages=de_DE.UTF-8
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION allow-external-password-cache
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-ok=_OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-cancel=_Cancel
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-yes=_Yes
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- ERR 83886254 Unknown option <Pinentry>
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-no=_No
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- ERR 83886254 Unknown option <Pinentry>
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-prompt=PIN:
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-pwmngr=_Save in password manager
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-cf-visi=Do you really want to make your passphrase visible on the screen?
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- ERR 83886254 Unknown option <Pinentry>
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-tt-visi=Make passphrase visible
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- ERR 83886254 Unknown option <Pinentry>
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION default-tt-hide=Hide passphrase
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- ERR 83886254 Unknown option <Pinentry>
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> OPTION touch-file=/Users/swiesner/.gnupg/S.gpg-agent
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> GETINFO pid
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- D 2017
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 -> INQUIRE PINENTRY_LAUNCHED 2017
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_6 <- END
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> SETKEYINFO n/9C318FEBE3DFD3EFE5980752F9E336BC31CAD7A0
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> SETDESC Please enter the passphrase to unlock the OpenPGP secret key:%0A%22Sebastian Wiesner (Social address) <me@lunaryorn.com>%22%0A4096-bit RSA key, ID 5C42FE98,%0Acreated 2016-04-11.%0A
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> SETPROMPT Passphrase:
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 <- OK
2016-05-07 13:02:10 gpg-agent[1978] DBG: chan_7 -> [[Confidential data not shown]]
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_7 <- [[Confidential data not shown]]
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_7 <- [[Confidential data not shown]]
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 30 120
2016-05-07 13:02:12 gpg-agent[1978] DBG: chan_6 -> S PROGRESS need_entropy X 120 120
2016-05-07 13:02:12 gpg-agent[1978] S2K calibration: 22369280 -> 100ms
```

**Comment by lunaryorn** [comment]
PS: Happens with the bottle as well as with a source build.  Couldn't try `--HEAD` because it fails to build for me.

I'm back to gpg2 for now.

**Comment by DomT4** [comment]
Unless this doesn't happen when GnuPG21 is built outside of Homebrew on OS X this is probably worth reporting upstream and seeing what Werner thinks. There have been several bugs with 2.1.x on OS X, so this could be a "legit" bug.

**Comment by lunaryorn** [comment]
@DomT4 Thanks, but how do I report issues to upstream here?  Can't you just forward the report?

I'm sorry but please understand that I'd rather prefer not to spend any more significant amount of time on this problem.  See, I've already spent a lot of time disproportionate to the significance of gpg in my tool chain and I'd rather just go back to a working version instead of pouring more work in here.

I'm sorry

**Comment by DomT4** [comment]
> Thanks, but how do I report issues to upstream here? Can't you just forward the report?

Werner may have questions that I can't answer on your behalf, not being able to read your mind 😄. 

Upstream's bug report system is at https://bugs.gnupg.org/gnupg/index if you wish to file a bug with them. Entirely up to you though. I'm afraid I'm personally still using `gnupg2` so I can't poke this much myself.

**Comment by gfontenot** [comment]
I'm seeing this too. I've created [a bug on the gnupg bug reporter](https://bugs.gnupg.org/gnupg/issue2358), but will be downgrading to 2.0 for now as well.

**Comment by lunaryorn** [comment]
I'd just like to say that the upstream bug has since been fixed, gnupg 2.1 is now working for me.

Thanks for your help, folks!
